src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1132,11 +1147,13 @@ private void WriteRedType(TreeType node)
                         {
                             if (i == 0)
                             {
-                                WriteLine("            return this.GetRedAtZero(ref this.{0});", CamelCase(field.Name));
+                                string suffix = IsOptional(field) ? "" : "!";
+                                WriteLine("            return this.GetRedAtZero(ref this.{0}){1};", CamelCase(field.Name), suffix);
                             }
                             else
                             {
-                                WriteLine("            return this.GetRed(ref this.{0}, {1});", CamelCase(field.Name), i);
+                                string suffix = IsOptional(field) ? "" : "!";
```

Comment:
why do we need `!` in the generated code?

Reply:
➡️ This is visible in 3f8ebc4f56a0ebd04868f7d02772fd2bfccc42be

Reply:
Warrants a comment though.

Reply:
Additionally, a bunch of these places do not require a suppression, though I think I'm ok with leaving them in to make generation easier.

Reply:
Actually, I'm not sure I understand why these are necessary. `GetRed` and `GetRedAtZero` are currently oblivious, so I'm not sure what warnings this would be suppressing.

Reply:
@333fred this branch was extracted from a branch where the methods were annotated. Once both are in place, the suppressions will not be optimal. The review was too overwhelming to send them all at once.